client/client.go

```diff
@@ -311,6 +311,9 @@ func ParseHostURL(host string) (*url.URL, error) {
 func (cli *Client) Dialer() func(context.Context) (net.Conn, error) {
 	return func(ctx context.Context) (net.Conn, error) {
 		if transport, ok := cli.client.Transport.(*http.Transport); ok {
+			if cli.scheme == "http" {
+				transport.TLSClientConfig = nil
+			}
```

Comment:
@AkihiroSuda I'm wondering what the root cause is of this; I had a quick look at some of the code, and looking at https://github.com/moby/moby/blob/0910306bf970603ce787466a98e4294ba81af841/client/client.go#L146-L158


That appears to change `scheme` to `https` if TLS Config was found; is something wrong with the initialisation code?